store/src/main/java/org/apache/rocketmq/store/CommitLog.java

```diff
@@ -1871,9 +1878,13 @@ public ByteBuffer encode(final MessageExtBatch messageExtBatch) {
                 this.msgBatchMemory.put((byte) topicLength);
                 this.msgBatchMemory.put(topicData);
                 // 17 PROPERTIES
-                this.msgBatchMemory.putShort(propertiesLen);
-                if (propertiesLen > 0)
+                this.msgBatchMemory.putShort((short) (propertiesLen + batchPropLen));
+                if (propertiesLen > 0) {
                     this.msgBatchMemory.put(messagesByteBuff.array(), propertiesPos, propertiesLen);
+                }
+                if (batchPropLen > 0) {
+                    this.msgBatchMemory.put(batchPropData, 0, batchPropLen);
+                }
```

Comment:
Just put properties of MessageExtBatch into msgBatchMemory. The original code takes the properties of a single message, so it does not exist.

Reply:
Sure.